src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -37,11 +37,9 @@
     @Query("select distinct pe from ProgrammingExercise pe where pe.id = :#{#exerciseId}")
     Optional<ProgrammingExercise> findByIdWithTestCases(@Param("exerciseId") Long id);
 
-    // TODO: Investigate - this should return the templateParticipation result including its feedbacks - but for some reason they can't be retrieved.
-    // Get an a programmingExercise with template and solution participation, each with the latest result
-    @EntityGraph(attributePaths = { "templateParticipation.results.feedbacks", "solutionParticipation.results.feedbacks" })
```

Comment:
EntityGraph seems to do an outer join. Even though the results are limited in the query, all results are returned.
There was no user impact because the client components filter out the latest result.